### DISCUSSION TITLE
有关于bot连接的问题

### DISCUSSION THREAD

**Discussion by yhzcake** [nonebot/discussions#441] [问答]
在nonebot启动的时候需要onebot向nonebot上报一次才可以连接bot,这导致我希望成功启动nonebot时通过getbot去send msg会报错
怎么样让nonebot能自动连上onebot?或者onebot需要循环上报激活nonebot?

**Comment by he0119** [comment]
推荐使用 [on_bot_connect](https://nonebot.dev/docs/advanced/runtime-hook#bot-%E8%BF%9E%E6%8E%A5%E5%A4%84%E7%90%86)。

参考用法 <https://github.com/he0119/CoolQBot/blob/1e501f10d8f296beb2088c598894548f1070a09e/src/plugins/morning/plugins/hello/__init__.py#L39-L55>

**Reply by yhzcake** [reply]
大致了解了
看起来如果onebot一直不上报仍旧无法触发定时任务
我的解决办法是bot启动时直接给onebot发个post,让onebot上报回复来触发链接
